src/features/berthOffer/BerthSwitchOfferContainer.tsx

```diff
@@ -0,0 +1,123 @@
+} from './components/notifications';
+import { CREATE_BERTH_SWITCH_OFFER_MUTATION } from './mutations';
+import {
+  CREATE_BERTH_SWITCH_OFFER,
+  CREATE_BERTH_SWITCH_OFFERVariables as CREATE_BERTH_SWITCH_OFFER_VARS,
+} from './__generated__/CREATE_BERTH_SWITCH_OFFER';
+import { BERTH_APPLICATIONS_QUERY } from '../applicationList/queries';
+import SelectBerthLease from './selectBerthLease/SelectBerthLease';
+
+const BerthSwitchOfferContainer = () => {
+  const { t, i18n } = useTranslation();
+  const routerQuery = useRouterQuery();
+  const history = useHistory();
+
+  const harborId = routerQuery.get('harbor') || '';
+  const applicationId = routerQuery.get('application') || '';
+
+  const [selectedBerth, setSelectedBerth] = useState<BerthData | undefined>(undefined);
+  const [showSelectLease, setShowSelectLease] = useState<boolean>(false);
+
+  const { loading: applicationLoading, error: applicationError, data: applicationData } = useQuery<
+    BERTH_OFFER,
+    BERTH_OFFER_VARS
+  >(BERTH_OFFER_QUERY, {
+    variables: { applicationId, servicemapId: harborId },
+  });
+
+  const [createBerthSwitchOfferMutation, { loading: isSubmitting }] = useMutation<
+    CREATE_BERTH_SWITCH_OFFER,
+    CREATE_BERTH_SWITCH_OFFER_VARS
+  >(CREATE_BERTH_SWITCH_OFFER_MUTATION, {
+    refetchQueries: [getOperationName(BERTH_APPLICATIONS_QUERY) || 'BERTH_APPLICATIONS'],
+  });
+  const createBerthSwitchOffer = (berth: BerthData, oldLeaseId?: string) =>
+    createBerthSwitchOfferMutation({
+      variables: {
+        input: {
+          applicationId: applicationId,
+          newBerthId: berth.id,
+          oldLeaseId: oldLeaseId,
+        },
+      },
+    });
+
+  if (applicationLoading) return <LoadingSpinner isLoading />;
+
+  const data = applicationData?.harborByServicemapId;
+  const boat = getApplicationBoat(applicationData?.berthApplication, applicationData?.boatTypes);
+  const customerId = applicationData?.berthApplication?.customer?.id;
+
+  if (!data) return <NoDataNotification />;
+  if (applicationError) return <ErrorNotification />;
+  if (!boat) return <NoBoatErrorNotification />;
+  if (!customerId) return <ErrorNotification />;
+
+  const application = applicationData?.berthApplication && {
+    date: formatDate(applicationData.berthApplication.createdAt, i18n.language),
+    status: applicationData.berthApplication.status,
+    type: t(getApplicationTypeTKey(!!applicationData.berthApplication.berthSwitch)),
+  };
+  const tableData = getBerthData(data);
+  const harbor = getHarbor(data);
+  const piersIdentifiers = getAllPiersIdentifiers(data?.properties?.piers);
+
+  const handleReturn = () => history.goBack();
+  const handleClickSelect = (berth: BerthData) => {
+    setSelectedBerth(berth);
+    createBerthSwitchOffer(berth)
+      .then((result) => {
+        if (result.errors) return;
+        history.goBack();
+        showBerthSuccessToast(berth, t);
+      })
+      .catch(() => {
+        setShowSelectLease(true);
```

Comment:
Why do we want to set this to `true` when we get an error?

Reply:
:derp: Indeed, good catch!

Reply:
Actually, this is for the case when the backend can't figure out what the existing lease is supposed to be. The admin then selects whichever they think is the correct one